py-polars/tests/unit/io/test_database_read.py

```diff
@@ -925,3 +926,21 @@ def test_read_database_async(tmp_sqlite_db: Path) -> None:
             execute_options=execute_opts,
         )
         assert_frame_equal(expected_frame, df)
+
+
+async def _nested_async_test(tmp_sqlite_db: Path) -> pl.DataFrame:
+    async_engine = create_async_engine(f"sqlite+aiosqlite:///{tmp_sqlite_db}")
+    return pl.read_database(
+        query="SELECT id, name FROM test_data ORDER BY id",
+        connection=async_engine.connect(),
+    )
```

Comment:
Can you explain what is 'nested' about this query? Looks like a regular query to me.

Reply:
It triggers an `asyncio.run` inside an existing `asyncio.run`, potentially nesting event loops; I just realised I could drastically simplify the synchronising handler (which converts async to sync and handles reentrant loops like this), so it should be a bit more obvious to see in the new/updated code.

Reply:
Thanks, I see it now.